terraform/lib/dependabot/terraform/file_updater.rb

```diff
@@ -18,14 +19,18 @@ def updated_dependency_files
         updated_files = []
 
         [*terraform_files, *terragrunt_files].each do |file|
-          next unless file_changed?(file)
+          next unless file_changed?(file) || lock_file?(file.name)
 
           updated_content = updated_terraform_file_content(file)
-          raise "Content didn't change!" if updated_content == file.content
+
+          raise "Content didn't change!" unless lockfile_changed? || updated_content != file.content
```

Comment:
Are these changes required? Looks like this `each` loop won't loop over any lockfiles now that the `terragrunt_files` method exlcudes these so wondering if we can revert the above changes and ✂️  the `lockfile_changed?` method?

Reply:
Gotcha but looks like this is guarded by `file_changed?` in line `22` so shouldn't get this far, as this block only loops over manifests. We then check if we have no updated files again on line `34` after the lockfile update so should cover a lockfile only update scenario.

Reply:
I had a poke locally and seems like tests pass without these changes. From my understanding we never loop over any lockfiles here so we're only checking if a manifest's requirements have changed. I remember seeing a past commit where the `each` block included lockfiles which would have required these changes.

Reply:
Originally, the `each` block looped over `*lock_files`--it no longer does, so the lockfile guards are no longer necessary.

We now attempt to update the lockfile outside of the `each` block irrespective of changes to the manifest file. Thus, if a new version satisfies the constraints of the manifest file, the manifest files will not not update but the lockfile still will.